rest/context.go

```diff
@@ -72,6 +72,25 @@ type RequestContext interface {
 	// string is returned with the error set.
 	NextURL() (string, error)
 
+	// BuildURL builds a full URL for the given resource name & method name with the
+	// key/value pairs for the route variables.
+	//
+	// resourceName should have the same value as the handler's ResourceName method.
+	//
+	// methodName should be the Handle* constant that corresponds with the resource
+	// method for which to build the URL. E.g. HandleCreate with build a URL that
+	// corresponds with the CreateResource method.
+	//
+	// pairs should be key/value pairs for any URL variables, passed on to
+	// gorilla's Route.URL method.
+	BuildURL(resourceName string, methodName string, pairs ...string) (string, error)
```

Comment:
See comment above about typedef for method name. Then this would look like:

```go
BuildURL(resourceName string, method HandleMethod, pairs ...string) (string, error)
```

It's more clear to the user what valid values of `methodName` are.